### PR TITLE
Desktop: Add `escape` to go back from Dropbox Login screen

### PR DESCRIPTION
**Pull request by devonzuegel**
**Problem:** I habitually press <kbd>⌘</kbd> + <kbd>S</kbd> in notes, which in development mode (i.e. where I, and likely others, don't have a sync target set up) takes you to the `Dropbox Login` screen. It's annoying to have to mouse over to the `Back` button on the top left.

**Solution:** I've added a keyboard shortcut to escape out of that screen: <kbd>esc</kbd>. 
Here's a demo with the key visualizer on:

### CODE REVIEW

**Comment by laurent22** [comment]
I'm a bit conflicted about this PR. It's useful, but:

- First if we add a shortcut to go back to the previous screen, then this should be applied to all the screens, not just to one.

- Secondly, I don't think escape is that common to go back to the previous screen? Some browsers use Backspace, other Ctrl+Left arrow, but escape not so much. But it's a detail I guess, compared to the first issue.

As it is, I feel we are adding special code to a screen, and almost nobody will use the feature because they won't know it's there, and because the shortcut is not necessarily intuitive. It's almost like an easter egg :-)

So basically I would be fine with it, but only if it was a global solution that worked in all the screens (without adding boilerplate code to all of them). I don't know if it would be easy to do thought.

**Comment by tessus** [comment]
> Secondly, I don't think escape is that common to go back to the previous screen?

I agree to disagree. Escape is used almost everywhere to cancel a transaction or to cancel a popup window, or as a shortcut for the button `Cancel`.

Fun fact: on MacBooks with TouchBar the esc key often changes to a cancel button.

**Comment by laurent22** [comment]
> I agree to disagree. Escape is used almost everywhere to cancel a transaction or to cancel a popup window, or as a shortcut for the button `Cancel`.

Yes that's true. I guess I had in mind the browser model because the app is made of web pages, but really it's a desktop app so we could go with ESC.

**Comment by tessus** [comment]
So in this case, we can merge this, right? Or is something still missing?

**Comment by laurent22** [comment]
> Or is something still missing?

Yes we'd still need to make this consistent, i.e. apply this change to all the screen.

**Comment by tessus** [comment]
Ah, ok. But in that case we should probably create a new PR for all screens. In the meantime there's no harm in merging this one, is there?
I can create a new PR, but it would help, if someone could list all the screens that need changes. (in a separate ticket). I can of course go through them myself, I just don't want to miss anything.

**Comment by laurent22** [comment]
> Ah, ok. But in that case we should probably create a new PR for all screens. In the meantime there's no harm in merging this one, is there?

No the issue is that if we want to add it to all the screens, we can't manually add the event handler code to all the screens, that would too much boilerplate code and difficult to maintain over time. We'll need to wrap the code somehow and make sure it's called from the relevant screens.

If we were using React hooks, we could use an effect for example. There's probably other ways to do it but not sure how, I'd need to look into it.

**Comment by tessus** [comment]
Thx for the info.

**Comment by laurent22** [comment]
I'll close this PR for the reasons mentioned above, but later we can consider implementing ESC to exit all the screens.

**Comment by tessus** [comment]
Do we have this tracked somewhere? Also, maybe this is a hacktoberfest issue. Someone with Electron and/or react knowledge should be able to implement this rather quickly.

(Wouldn't a react hook affect the mobile code as well?)

**Comment by laurent22** [comment]
Not sure if this is such an easy issue, I wouldn't even know how to do it right now as we don't have anything in place to share a behaviour among screens. I think that would be the first step, then we can look at adding ESC to exit screens and other shared behaviour.

Ideally I'd like something like we set a property "enableExitShortcut" on the component and all the components with that property set to `true` will get that behaviour automatically (without manually adding and removing event handlers).

**Comment by tessus** [comment]
Hmm, isn't this were OO comes into play? We add that exit method to the super class and it should be available to all screens.
Well, I don't know the current code, or maybe I'm missing something in this Electron/React/Reducer framework, but this is how it is supposed to work. At least I hope so, otherwise OO and frameworks don't make much sense.

**Comment by laurent22** [comment]
Yes but React components are special and base components are usually wrapped rather than extended.
